query-engine/connector-test-kit/src/test/scala/queries/relations/UnnecessaryDBRequests.scala

```diff
@@ -0,0 +1,258 @@
+package queries.relations
```

Comment:
This test effectively does nothing and I suggest removing the entire file. It was probably important for your investigation, but adds no value beyond that.

Reply:
Yep, that was the plan. This is only there to help me check the generated SQL, I can't think of an easy way to verify the generated statements from the Scala tests anyways.

Reply:
I talked a bit with Marcus and we decided to assert the number of selects being run to use this as a minimal regression test.